include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+     * conditions for you by using the `ic_type` parameter at construction time.
+     *
+     * You have three options
+     * -  none: do not try to make initial conditions consistent.
+     * -  use_y_diff: compute the algebraic components of y and differential
+     *    components of y_dot, given the differential components of y.
+     *    This option requires that the user specifies differential and
+     *    algebraic components in the function get_differential_components.
+     * -  use_y_dot: compute all components of y, given y_dot.
+     *
+     * By default, this class assumes that all components are differential, and that you want
+     * to solve a standard ode. In this case, the initial component type is set to `use_y_diff`,
+     * so that the `y_dot` at time t=`initial_time` is computed by solving the nonlinear problem
+     * $F(y_dot, y(t0), t0) = 0$ in the variable `y_dot`.
+     *
+     * Notice that a Newton solver is used for this computation. The Newton solver parameters
+     * can be tweaked by acting on `ic_alpha` and `ic_max_iter`.
+     *
+     * If you reset the solver at some point, you may want to select a different computation
+     * for the initial conditions after reset. Say, for example, that you have refined a grid,
+     * and after transferring the solution to the new grid, the initial conditions are no longer
+     * consistent. Then you can choose how these are made consistent, using the same three
+     * options that you used for the initial conditions in `reset_type`.
+     *
+     * @param mpi_comm MPI communicator
+     * @param initial_time Initial time
+     * @param final_time Final time
+     * @param initial_step_size Initial step size
+     * @param min_step_size Minimum step size
+     * @param abs_tol Absolute error tolerance
+     * @param rel_tol Relative error tolerance
+     * @param max_order Maximum BDF order
+     * @param output_period Time interval between each output
+     * @param ignore_algebraic_terms_for_errors Ignore algebraic terms for error computations
+     * @param ic_type Initial condition type
+     * @param reset_type Initial condition type after restart
+     * @param ic_alpha Initial condition Newton parameter
+     * @param ic_max_iter Initial condition Newton max iterations
+     * @param max_non_linear_iterations Maximum number of nonlinear iterations
+     * @param verbose Show output of time steps
+     * @param use_local_tolerances Use local tolerances
+     */
+    IDA(const MPI_Comm mpi_comm = MPI_COMM_WORLD,
+        const double &initial_time = 0.0,
+        const double &final_time = 1.0,
+        const double &initial_step_size = 1e-4,
+        const double &min_step_size = 1e-6,
+        const double &abs_tol = 1e-6,
+        const double &rel_tol = 1e-5,
+        const unsigned int &max_order = 5,
+        const double &output_period = .1,
+        const bool &ignore_algebraic_terms_for_errors = true,
+        const std::string &ic_type = "none",
```

Comment:
i would this and `reset_type`to enumerators and only internally switch to `std::string` to provide the correct input to Sundials. Same for class private variables.